database/user.go

```diff
@@ -225,25 +256,40 @@ func (db *DB) UserByStripeID(ctx context.Context, id string) (*User, error) {
 }
 
 // UserCreate creates a new user in the DB.
-func (db *DB) UserCreate(ctx context.Context, sub string, tier int) (*User, error) {
-	// Check for an existing user with this sub.
-	users, err := db.managedUsersByField(ctx, "sub", sub)
+func (db *DB) UserCreate(ctx context.Context, email, pass, sub string, tier int) (*User, error) {
+	// Check for an existing user with this email.
+	users, err := db.managedUsersByField(ctx, "email", email)
 	if err != nil && !errors.Contains(err, ErrUserNotFound) {
 		return nil, errors.AddContext(err, "failed to query DB")
 	}
 	if len(users) > 0 {
 		return nil, ErrUserAlreadyExists
 	}
-	_, email, err := jwt.UserDetailsFromJWT(ctx)
-	if err != nil {
-		// Log the error but don't do anything differently.
-		db.staticLogger.Debugf("We failed to extract the expected user infotmation from the JWT token. Error: %s", err.Error())
+	// Check for an existing user with this sub.
+	if sub != "" {
+		users, err := db.managedUsersByField(ctx, "sub", sub)
+		if err != nil && !errors.Contains(err, ErrUserNotFound) {
+			return nil, errors.AddContext(err, "failed to query DB")
+		}
+		if len(users) > 0 {
+			return nil, ErrUserAlreadyExists
+		}
+	} else {
+		sub = generateSub()
+	}
+	var passHash []byte
+	if pass != "" {
```

Comment:
Is no password accepted? Shouldn't that be an error condition?

Reply:
Comment explaining why not?

Reply:
User management was handled in Kratos until now, so there were several places where we create an `accounts` user based on the passed JWT token, which only contains an email and a sub.

For the moment I am going with the safer assumption which is that we'll be running kratos and the new accounts system together, so it would still be possible for people to create accounts with kratos and get them registered with `accounts`. The bolder assumption would be that Kratos will be entirely gone and accounts will be the only thing managing users. But having in mind that we deploy as a rolling wave, we'll have a heterogeneous system at first, so assuming Kratos still exists seems like the logical thing to do. We'll clean up these parts in a subsequent PR.

I added a comment explaining that.